src/scp/BallotProtocol.cpp

```diff
@@ -2009,6 +2011,64 @@ BallotProtocol::getJsonInfo()
     return ret;
 }
 
+SCP::QuorumInfoNodeState
+BallotProtocol::getState(NodeID const& n, bool selfAlreadyMovedOn)
+{
+    if (n == mSlot.getLocalNode()->getNodeID())
+    {
+        // always mark myself as IN_PROGRESS.
+        return SCP::QuorumInfoNodeState::IN_PROGRESS;
+    }
+    auto state = SCP::QuorumInfoNodeState::IN_PROGRESS;
+
+    auto it = mLatestEnvelopes.find(n);
+    if (it == mLatestEnvelopes.end())
+    {
+        if (mHasTimerExpired || selfAlreadyMovedOn)
+        {
+            state = SCP::QuorumInfoNodeState::MISSING;
+        }
+        else
+        {
+            // It's too soon to start calling this node
+            // MISSING.
+            state = SCP::QuorumInfoNodeState::NO_INFO;
+        }
+    }
+    else if (mCurrentBallot)
```

Comment:
hmm, should we only use local state that's been fully validated? Otherwise you might be potentially comparing slots with bogus votes, and I don't think that's correct